feat(content): adding deep link docs (#14006)
newrelic/docs-website: src/content/docs/style-guide/formatting/hyperlinks.mdx

* When reorganizing a doc, try to preserve incoming anchor links where possible. Be mindful that when you delete a level two heading `##`, it will break any links to that heading.
* Avoid changing an existing anchor ID, because we may be monitoring traffic to it from our internal New Relic account.

## Deep links into product [#deep-links]

> When directing users to parts of the product, you want to deep link when possible. Deep linking encourages users to go into the product and, if they haven't created an account yet, prompts them to sign up. This is more effective than just linking to a sign up page. However, deep linking into the UI only works sometimes and it takes some experimenting to find a good permalink. 

Reviewer: @akristen Would it be helpful to define what a deeplink is and provide an example link?

Reviewer 2: @akristen  +1 to add an example

> Here's an example workflow using the Logs UI as an example: 

Author: ```suggestion
Here's an example workflow using the Logs UI: 
```

1. From your work account, go to the part of the UI you want to link. Grab the complete URL as it appears naturally.
2. Open an incognito window and paste in the URL.

> 3. Remove all parts of the URL. We want simple URLs when possible. For logs, this looks like this: `https://one.newrelic.com/logger`. 

Author: ```suggestion
3. Edit the URL to a cleaner state. We want simple URLs when possible. For logs, this looks like this: `https://one.newrelic.com/logger`. 
```

> 3. Remove all parts of the URL. We want simple URLs when possible. For logs, this looks like this: `https://one.newrelic.com/logger`. 

Reviewer: @akristen I know it's tough to do this in a procedure, but it might be good to do a URL breakdown. Not as specific as this, but perhaps calling out some important parts: https://www.lucamussari.com/wp-content/uploads/2021/06/parts-of-a-url-luca-mussari-com.png

> 4. It'll prompt you to sign in. Sign in with a non-work email. The URL sends you to the logs page and auto-completes account ID identifying information. That page is deep linkable.

Author: ```suggestion
4. New Relic will prompt you to sign in, so sign in with a non-work email. 

The URL sends you to the logs page and auto-completes account ID identifying information. That page is deep linkable.
```

> If the simple URL doesn't work, you have two extra steps. Here's an example workflow where the simple URL doesn't work. We'll use this URL to work off of: 

Author: ```suggestion
If the simple URL doesn't work, you some extra steps. We'll use this URL to work off of: 
```

```

> https://one.newrelic.com/marketplace/install-data-source?account=[ACCOUNT_ID_HERE]&begin=168917[…]APPLICATION%27%29&state=1ecc7248-ff69-a838-0c8e-d5eaa8bc5431

Reviewer: @akristen The [...] in that URL makes me think that the URL isn't complete...

```

> 1. After finding your URL, open an incognito window and paste in the simple URL: `https://one.newrelic.com/marketplace/install-data-source`This page doesn't load with the simplified URL. 

Author: ```suggestion
1. After finding your URL, open an incognito window and paste in the simple URL: `https://one.newrelic.com/marketplace/install-data-source` This page won't load with the simplified URL. 
```

2. Test this URL by removing the `account=[ACCOUNT_ID_HERE]` part of the slug. The new URL will look like this: 

```
https://one.newrelic.com/marketplace/install-data-source?&begin=168917[…]APPLICATION%27%29&state=1ecc7248-ff69-a838-0c8e-d5eaa8bc5431
```
3. Paste that URL in a new incognito window. It'll take you to the correct page with your unique account ID populated in.

> 4. Confirm with another writer that this URL works by having them open the URL with the account ID number removed.  

Author: ```suggestion
4. Confirm with another writer that this URL works by having them open the URL with the account ID number removed.  They should use a non-work email.
```
